File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/SubstanceProtein.ts
import { Attachment } from "./Attachment";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type SubstanceProteinSubunit = BackboneElement & {
    length?: number;
    subunit?: number;
    sequence?: string;
    sequenceAttachment?: Attachment;
    cTerminalModification?: string;
    nTerminalModification?: string;
    cTerminalModificationId?: Identifier;
    nTerminalModificationId?: Identifier;
};

export type SubstanceProtein = DomainResource & {
    subunit?: SubstanceProteinSubunit[];
    sequenceType?: CodeableConcept;
    disulfideLinkage?: string[];
    numberOfSubunits?: number;
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/SubstanceProtein.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Attachment
from base import CodeableConcept
from base import DomainResource
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class SubstanceProteinSubunit(BackboneElement):
    length: Optional[int] = None
    subunit: Optional[int] = None
    sequence: Optional[str] = None
    sequence_attachment: Optional[Attachment] = None
    c_terminal_modification: Optional[str] = None
    n_terminal_modification: Optional[str] = None
    c_terminal_modification_id: Optional[Identifier] = None
    n_terminal_modification_id: Optional[Identifier] = None

@dataclass(kw_only=True)
class SubstanceProtein(DomainResource):
    subunit: Optional[List[SubstanceProteinSubunit]] = field(default_factory=list)
    sequence_type: Optional[CodeableConcept] = None
    disulfide_linkage: Optional[List[str]] = field(default_factory=list)
    number_of_subunits: Optional[int] = None